### PR TITLE
fix(ttlcache): goroutine leak

### PR DESCRIPTION
**Pull request by shhdgit**
Related: [pd/pull/3604](https://github.com/tikv/pd/pull/3604)

Cleanup ttl cache at end of lifecycle.

### CODE REVIEW

**Comment by shhdgit** [comment]
I'm not sure this will solve the problem https://github.com/tikv/pd/pull/3604/checks?check_run_id=2375055816. Maybe we should also add goleak testing.
@breeswish @baurine  PTAL.

File: pkg/utils/sys_schema.go [hunk 1]
```diff
@@ -29,11 +29,9 @@ type SysSchema struct {
 	cache *ttlcache.Cache
 }
 
-func NewSysSchema() *SysSchema {
-	c := ttlcache.NewCache()
-	c.SkipTTLExtensionOnHit(true)
+func NewSysSchema(cache *ttlcache.Cache) *SysSchema {
 	return &SysSchema{
-		cache: c,
+		cache: cache,
```
Comment (breeswish) [inline pkg/utils/sys_schema.go:34]: This may be wrong since multiple NewSysSchema will share the same cache and cause cache entity conflicts?

Comment (shhdgit) [reply pkg/utils/sys_schema.go:34]: Introduce lifecycle hooks is a good idea.

**Review by breeswish — APPROVED** [review]
Looks much nicer!

**Pull request merged**